source/server-discovery-and-monitoring/server-monitoring.rst

```diff
@@ -91,6 +91,28 @@ Round trip time. The client's measurement of the duration of one hello or legacy
 The RTT is used to support `localThresholdMS`_ from the Server Selection spec
 and `timeoutMS`_ from the `Client Side Operations Timeout Spec`_.
 
+FaaS
+````
+
+A Function-as-a-Service (FaaS) environment like AWS Lambda.
+
+serverMonitoringMode
+````````````````````
+
+The serverMonitoringMode option configures which server monitoring protocol to use. Valid modes are
+"stream", "poll", or "auto". The default value MUST be "auto":
+
+- With "stream" mode, the client MUST use the streaming protocol when the server supports
```

Comment:
Is there a reason we don't throw an error when a user explicitly requests the streaming protocol but is using an incompatible server?

Reply:
There are a few reasons:
- "stream" mode is the same as how driver's work today. Throwing an error would introduce even more new behavior.
- Better compatibility. It's theoretically possible for the server to dynamically disable streaming support in which case we'd want the drivers to keep working.
- If a user really wants to raise an error in this case they can easily check for streaming support themselves.

Reply:
Compatibility is something I hadn't considered, thanks.